### ISSUE TITLE
Background view borked

### ISSUE THREAD

**Issue by UltraBlackLinux** [WayfireWM/wayfire-plugins-extra#154]
Hey there,
I just checked, and was able to confirm, that the background-view plugin is no longer working. The binary entered into the text field is not executed. The logs didn't tell me about any errors.
Can you reproduce this?
I'm on `0.8.0-eaf81e30 (Nov 30 2022, branch 'fix-refocus-logic')`
Thanks!

**Comment by soreau** [comment]
You must also set the [app-id](https://github.com/WayfireWM/wayfire-plugins-extra/blob/master/metadata/background-view.xml#L15).

**Comment by UltraBlackLinux** [comment]
As previously determined, only `nil` works - anything else does not work, or did not work at least.

**Comment by soreau** [comment]
If the app does not set an id, it will match `nil`, yes. Is this a problem for some reason?

**Comment by UltraBlackLinux** [comment]
This worked for months and I doubt that it just stops working.

**Comment by soreau** [comment]
I cannot reproduce with latest wayfire-plugins-extra f6a083c2. Maybe you need to rebuild latest wayfire-plugins-extra?

**Comment by UltraBlackLinux** [comment]
Just gave that a shot - still nothing, sadly.

Even tried the default, mpv, and that also doesn't launch.

**Comment by soreau** [comment]
Can you upload the log when running with `wayfire -d`?

**Comment by UltraBlackLinux** [comment]
[wf.txt](https://github.com/WayfireWM/wayfire-plugins-extra/files/10142609/wf.txt)

Noting special, at least I didn't see anything. At the end I toggled the background view on and off

**Comment by soreau** [comment]
I see a few messages like these:
```
EE 02-12-22 17:12:02.884 - [src/output/plugin-loader.cpp:185] Failed to load plugin "background-view". Make sure it is installed in /usr/local/lib/wayfire or in $WAYFIRE_PLUGIN_PATH.
```
It looks like you have wayfire installed to `/usr/local/`. Maybe you are installing wayfire-plugins-extra to a different prefix, such as `/usr/`? Wayfire and the extra plugins should be installed to the same prefix.

**Comment by UltraBlackLinux** [comment]
> It looks like you have wayfire installed to `/usr/local/`. Maybe you are installing wayfire-plugins-extra to a different prefix, such as `/usr/`? Wayfire and the extra plugins should be installed to the same prefix.

No, never installed it there. I'm using the AUR wayfire-git package, which installes to /usr/bin and so on. Same paths for wayfire-plugins-extra-git

This is probably also why wf-kill stopped working. I wanted to open an issue on that, but forgot. This all seems related.

The plugins folder seems populated, and the xml files also all exist.

![screenshot-20221202-190040](https://user-images.githubusercontent.com/62404294/205356518-a3f832bd-e451-4c74-8bad-7e0c349e5825.png)

**Comment by UltraBlackLinux** [comment]
> Right, but apparently have an installation in /usr/local that is messing things up. If `which wayfire` says `/usr/local/bin/wayfire`, then you need to remove wayfire (and associated libs, like wf-config and plugins) from /usr/local.

Found nothing there interestingly.

What I did find out was, that it is related to my config for some reason. Maybe the parsing errors are the issue? I moved my config and it magically worked. Maybe you can spot what's wrong with mine?

[Uploading wayfire.ini.txt…]()

**Comment by UltraBlackLinux** [comment]
Now with the invisible existing wayfire installation in /usr/local, wayfire now crashes. It's trying to find libwlroots.so.10 while only .11 is avaiable - Why that now? I just compiled it, so shouldn't it use the correct version? Weird

**Comment by UltraBlackLinux** [comment]
Well. After a LOT of trouble and issues with crashes and such, I got it back to a working state. Thank you so much for your help! I wouldn't have noticed that issue if you hadn't told me!

**Comment by soreau** [comment]
FWIW, the one thing I notice in your config file are commas instead of decimal points for double values. This is probably a wf-config bug, since it should have been fixed by wf-config [a5adfda0](https://github.com/WayfireWM/wf-config/commit/a5adfda0d5), but it should be fixed.
